seqr/views/utils/dataset_utils.py

```diff
@@ -449,3 +450,115 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
         logger.warning(warning, user)
 
     return samples_to_load, info, warnings
+
+
+TOOL_FIELD = 'tool'
+PHENOTYPE_PRI_HEADER = {
+    'tool': TOOL_FIELD, 'project': 'project', 'sampleId': 'sample_id', 'rank': 'rank', 'geneId': 'gene_id',
+    'diseaseId': 'disease_id', 'diseaseName': 'disease_name', 'scoreName1': 'score_name1', 'score1': 'score1',
+    'scoreName2': 'score_name2', 'score2': 'score2', 'scoreName3': 'score_name3', 'score3': 'score3'}
+
+
+def _parse_phenotype_pri_row(row):
+    record = {mapped_key: row[key] for key, mapped_key in PHENOTYPE_PRI_HEADER.items()}
+
+    tool = PhenotypePrioritization.TOOL_LOOKUP.get(record[TOOL_FIELD], None)
+    if not tool:
+        raise ValueError('Expecting {} for the "tool" column but found {}'.format(
+            ', '.join([v for k, v in PhenotypePrioritization.TOOL_CHOICES]), record[TOOL_FIELD]))
+    record[TOOL_FIELD] = tool
+
+    scores = {record.pop('score_name1'): record.pop('score1')}
+    for score_index in ['2', '3']:
+        score_name = record.pop('score_name' + score_index, None)
+        score = record.pop('score' + score_index, None)
+        if score_name:
+            scores[score_name] = score
+    record['scores'] = scores
+
+    return record
+
+
+def _load_phenotype_pri_file(file_path):
+    data_by_id = defaultdict(dict)
+    f = file_iter(file_path)
+    header = _parse_tsv_row(next(f))
+    missing_cols = [col for col in PHENOTYPE_PRI_HEADER.keys() if col not in header]
+    if missing_cols:
+        raise ValueError(f'Invalid file: missing column(s) {", ".join(missing_cols)}')
+
+    count = 0
+    for line in tqdm(f, unit=' rows'):
+        row = dict(zip(header, _parse_tsv_row(line)))
+        record = _parse_phenotype_pri_row(row)
+        sample_id = record.pop('sample_id', None)
+        project = record.pop('project', None)
+        if not sample_id or not project:
+            raise ValueError('Both sample ID and project fields are required.')
+        if data_by_id[sample_id]:
+            if project != data_by_id[sample_id]['project']:  # a sample must belong to a single project
```

Comment:
sample IDs are not unique across projects, and I see no reason why we shouldn't let you include the same sample ID for multiple projects in the same file. I think you should change around the structure of how you are storing this so it is a dictionary keyed by project, and then the values are a dictionary keyed by sample with a list of records. So you would have your structure be ` defaultdict(lambda: defaultdict(list))` and then all you would do here is 
```
data_by_id[project][sample_id].append(record)
```

Reply:
Good to know that.